setup.py

```diff
@@ -4,34 +4,148 @@
+import sys
+from io import open
 from os import path
-from codecs import open
-from setuptools import setup, Extension
-
-description = open(path.join(path.abspath(path.dirname(__file__)),
-                             'README.rst'), encoding='utf-8').read()
-
-setup( name        = 'freetype-py',
-       version     = '1.2',
-       description = 'Freetype python bindings',
-       long_description = description,
-       author      = 'Nicolas P. Rougier',
-       author_email= 'Nicolas.Rougier@inria.fr',
-       url         = 'https://github.com/rougier/freetype-py',
-       packages    = ['freetype', 'freetype.ft_enums'],
-       classifiers = [
-          'Development Status :: 5 - Production/Stable',
-          'Environment :: X11 Applications',
-          'Environment :: MacOS X',
-          'Intended Audience :: Developers',
-          'License :: OSI Approved :: GNU General Public License (GPL)',
-          'Operating System :: MacOS',
-          'Operating System :: Microsoft :: Windows',
-          'Operating System :: POSIX',
-          'Operating System :: Unix',
-          'Programming Language :: Python :: 2',
-          'Programming Language :: Python :: 3',
-          'Topic :: Multimedia :: Graphics',
-          ],
-       keywords    = ['freetype', 'font'],
-     )
+
+from setuptools import Extension, find_packages, setup
+from setuptools.command.build_ext import build_ext
+
+cmdclass = {}
+try:
+    from wheel.bdist_wheel import bdist_wheel
+except ImportError:
+    print("warning: wheel package is not installed", file=sys.stderr)
+else:
+
+    class UniversalBdistWheel(bdist_wheel):
+        def get_tag(self):
+            return (
+                'py2.py3',
+                'none',
+            ) + bdist_wheel.get_tag(self)[2:]
+
+    cmdclass['bdist_wheel'] = UniversalBdistWheel
+
+
+class SharedLibrary(Extension):
+    """Object that describes the library (filename) and how to make it.
+
+    Note: The `output_dir` parameter shouldn't include the final lib/ or bin/
+    component, we figure that out on the fly.
+    """
+    if sys.platform == "darwin":
+        prefix = "lib"
+        suffix = ".dylib"
+        lib_dir = "bin"
+    elif sys.platform == "win32":
+        prefix = ""  # VC prefers to leave out the "lib" prefix.
+        suffix = ".dll"
+        lib_dir = "bin"  # VC likes to put DLLs into PREFIX/bin instead.
+    else:
+        prefix = "lib"
+        suffix = ".so"
+        lib_dir = "lib"
+
+    def __init__(self, name, cmd, cwd=".", output_dir=".", env=None):
+        Extension.__init__(self, name, sources=[])
+        self.cmd = cmd
+        self.cwd = path.normpath(cwd)
+        self.output_dir = path.join(path.normpath(output_dir), self.lib_dir)
+        self.env = env or dict(os.environ)
+
+
+class SharedLibBuildExt(build_ext):
+    """Object representing command to produce and install a shared library."""
+
+    # Apparently needed for determining the correct library filename on
+    # Windows.
```

Comment:
remove this comment. that's not only needed for windows; it's the thing that makes it possible to make setuptools and wheel to believe that this module contains a native extension module (whereas in reality it contains a shared library).